src/codegen/forward_demand.jl

```diff
@@ -185,6 +185,8 @@ function forward_visit!(ir::IRCode, ssa::SSAValue, order::Int, ssa_orders::Vecto
         recurse(stmt)
     elseif isexpr(stmt, :code_coverage_effect)
         return
+    elseif isexpr(stmt, :loopinfo) #TODO preserve this properly
```

Comment:
It's not legal to preserve.

Reply:
It can't be outright preserved but we might be able to do *something* with this infromation